dashboard/app/models/level.rb

```diff
@@ -239,6 +239,14 @@ def strip_name
     self.name = name.to_s.strip unless name.nil?
   end
 
+  # Find the first script this level belongs to that is a HOC script.
+  # Returns nil if this level does not belong to any HOC scripts.
+  def find_hoc_script
+    hoc_script_level = self.script_levels.find{|l| l.script.hoc?}
```

Comment:
oh wait I'm wrong, we would need to mess with the Script cache to preload the levels -> script_levels association...